components/merkledb/src/views/metadata.rs

```diff
@@ -284,7 +328,8 @@ impl<T: RawAccess> IndexesPool<T> {
     {
         let len = self.len();
         let metadata = IndexMetadata {
-            identifier: len,
+            identifier: len + 1,
+            // ^-- Identifier should be non-zero to translate to a correct id in `ResolvedAddress`
```

Comment:
I still don't quire understand what causes this requirement?

Reply:
`NonZeroU64` is used as the type of the identifier, together with `None` option reserved for non-index views (e.g., `__METADATA__`). Hence, the index ID should be, well, non-zero.